pkg/controller/apischeme/apischeme_controller_test.go

```diff
@@ -29,3 +44,263 @@ func TestClusterBaseDomain(t *testing.T) {
+			ClientErr:     map[string]string{"on": "Get", "type": "InternalError", "target": "*v1.Service"},
+		},
+		{
+			Name:          "Should requeue without error when successfully removing admin API A (DNS) record",
+			Resp:          reconcile.Result{Requeue: true},
+			ErrorExpected: false,
+			ClientObj:     []testutils.ClientObj{apischemeWithDeleteTimestampWithFinalizer},
+			RuntimeObj:    []testutils.RuntimeObj{infraObj, nlbService},
+			CloudMockFunc: mockCloudExpectDeleteAdminAPIDNS,
+			CloudMockArgs: cloudmockargs{
+				cloud: cloud,
+				cobj:  apischemeWithDeleteTimestampWithFinalizer.Obj,
+				robj:  nlbService.Obj,
+			},
+		},
+		{
+			Name:          "Should requeue with delay and without error when LB is not yet ready",
+			Resp:          reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second},
+			ErrorExpected: false,
+			ClientObj:     []testutils.ClientObj{apischemeWithDeleteTimestampWithFinalizer},
+			RuntimeObj:    []testutils.RuntimeObj{infraObj, nlbService},
+			CloudMockFunc: mockCloudExpectDeleteAdminAPIDNS,
+			CloudMockArgs: cloudmockargs{
+				cloud: cloud,
+				cobj:  apischemeWithDeleteTimestampWithFinalizer.Obj,
+				robj:  nlbService.Obj,
+				err:   cioerrors.NewLoadBalancerNotReadyError(),
+			},
+		},
+		{
+			Name:          "Should error when failing to update the DNS Record",
+			Resp:          reconcile.Result{},
+			ErrorExpected: true,
+			ErrorReason:   "",
+			ClientObj:     []testutils.ClientObj{apischemeWithDeleteTimestampWithFinalizer},
+			RuntimeObj:    []testutils.RuntimeObj{infraObj, nlbService},
+			CloudMockFunc: mockCloudExpectDeleteAdminAPIDNS,
+			CloudMockArgs: cloudmockargs{
+				cloud: cloud,
+				cobj:  apischemeWithDeleteTimestampWithFinalizer.Obj,
+				robj:  nlbService.Obj,
+				err:   cioerrors.NewDNSUpdateError("bad"),
+			},
+		},
+	}
+
+	for _, test := range tests {
+		testClient, testScheme := testutils.SetUpTestClient(test.ClientObj, test.RuntimeObj, test.ClientErr["on"], test.ClientErr["type"], test.ClientErr["target"])
+
+		if test.CloudMockFunc != nil {
+			test.CloudMockFunc(testClient, test.CloudMockArgs.cloud, test.CloudMockArgs.cobj, test.CloudMockArgs.robj, test.CloudMockArgs.err)
+		}
+
+		r := &ReconcileAPIScheme{client: testClient, scheme: testScheme}
+		result, err := r.Reconcile(context.TODO(), reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name:      name,
+				Namespace: namespace,
+			},
+		})
+
+		if err == nil && test.ErrorExpected || err != nil && !test.ErrorExpected {
+			t.Fatalf("Test [%v] return mismatch. Expect error? %t: Return %+v", test.Name, test.ErrorExpected, err)
+		}
+		if err != nil && test.ErrorExpected && test.ErrorReason != fmt.Sprint(k8serr.ReasonForError(err)) {
+			t.Fatalf("Test [%v] FAILED. Excepted Error %v. Got %v", test.Name, test.ErrorReason, k8serr.ReasonForError(err))
+		}
+		if result != test.Resp {
+			t.Fatalf("Test [%v] FAILED. Excepted Response %v. Got %v", test.Name, test.Resp, result)
+		}
```

Comment:
does that mean that only one error is displayed at a time while running these tests? or would it show the whole suite

this might be related to the fact all of the code is run in one main `TestReconcile`

Reply:
also, this is copied as it's clearer to read this way? I am fine with that but just throwing it out there